api/patients.ts

```diff
@@ -0,0 +1,80 @@
+import type { FetchOptions } from "@/hooks/datagrids/use-datagrid";
+import { http } from "@/lib/axios";
+import getTableSearchParams from "@/lib/get-search-params";
+import type { Patient } from "@/lib/types";
+import { notifications } from "@mantine/notifications";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { type AxiosResponse, isAxiosError } from "axios";
+import * as z from "zod";
+import { imageSchema } from "./common";
+
+export const createPatientSchema = z.object({
+	national_id: z.string().min(14).max(14),
+	full_name: z.string().min(3),
+	image: imageSchema.optional(),
+	phone: z.object({
+		mobile: z.preprocess(
+			(val) => val || undefined,
+			z.string().min(3).optional(),
+		),
+	}),
+	gender: z.literal("male").or(z.literal("female")).optional(),
+	address: z.object({
+		city: z.string().optional(),
+		governorate: z.string().optional(),
+		street: z.string().optional(),
+	}),
+	martial_status: z.string().optional(),
+	date_of_birth: z.date().optional(),
+	notes: z.preprocess((val) => val || undefined, z.string().min(3).optional()),
+});
+
+export async function createPatient(data: z.infer<typeof createPatientSchema>) {
+	return await http.post<Patient>("/accounts/patient/", data);
+}
+
+export function useCreatePatient({
+	onSuccess,
+}: {
+	onSuccess?: (response: AxiosResponse<Patient>) => void;
+} = {}) {
+	return useMutation({
+		mutationFn: createPatient,
+		onSuccess: (response) => {
+			notifications.show({
+				title: "Patient was added successfully",
+				message: "",
+				color: "white",
+				style: { backgroundColor: "#22c55e" },
+			});
+			onSuccess?.(response);
```

Comment:
what is the role of this line

Reply:
Invoke the `onSuccess` function. But onSuccess may be undefined so we use the cool feature of JavaScript `?.`.

Reply:
I think, there is no need to call it if the mutation is successful it will call automatically.